numpy/core/tests/test_multiarray.py

```diff
@@ -3575,20 +3575,19 @@ def collect_warning_types(f, *args, **kwargs):
 
         # All the different functions raise a warning, but not an error, and
         # 'a' is not modified:
-        assert_equal(collect_warning_types(a[['f1', 'f2']].__setitem__, 0, (10, 20)),
+        assert_equal(collect_warnings(a[['f1', 'f2']].__setitem__, 0, (10, 20)),
```

Comment:
assert_warns is probably a bit nicer (no need for the local collect warnings), but we can put that off to another time.